src/distribution/triangular.rs

```diff
@@ -0,0 +1,241 @@
+use distribution;
+use source::Source;
+
+/// An triangualr distribution.
+#[derive(Clone, Copy)]
+pub struct Triangular {
+    a: f64,
+    b: f64,
+    c: f64,
+}
+
+impl Triangular {
+    pub fn new(a: f64, b: f64, c: f64) -> Self {
+        should!(a < b && a <= c && c <= b);
+        Triangular {
+            a: a,
+            b: b,
+            c: c,
+        }
+    }
+}
+
+impl distribution::Continuous for Triangular {
+    #[inline]
+    fn density(&self, x: f64) -> f64 {
+        use std::f64;
+        use std::cmp::Ordering::*;
+        if x <= self.a {
+            0.0
+        } else if self.b <= x {
+            1.0
+        } else {
+            let diff = self.b - self.a;
+            match x.partial_cmp(&self.c) {
+                Some(Less) | Some(Equal) =>
+                    (x - self.a).powi(2) / diff / (self.c - self.a),
+                Some(Greater) =>
+                    1.0 - (self.b - x).powi(2) / diff / (self.b - self.c),
```

Comment:
According to the [style guide](https://github.com/rust-lang/rust/blob/master/src/doc/style/style/whitespace.md), one can use up to 99 characters per line. It should probably fit on one line.